mesos-client/src/main/scala/com/mesosphere/mesos/client/Session.scala

```diff
@@ -52,6 +54,28 @@ case class Session(url: URL, streamId: String, authorization: Option[Credentials
         Flow[Array[Byte]].map(createPostRequest(_, None)).via(connection)
     }
 
+  private def poolSettings():Future[ConnectionPoolSettings] = Future {
+    // Constructs the connection pool settings with defaults and overrides the max connections and pipelining limit so
+    // that only one request at a time is processed. See https://doc.akka.io/docs/akka-http/current/configuration.html
+    // for details.
+    // *IMPORTANT*: DO NOT CHANGE maxConnections OR pipeliningLimit! Otherwise, USI won't guarantee request order to Mesos!
+    ConnectionPoolSettings("").withMaxConnections(1).withPipeliningLimit(1)
+  }(ExecutionContext.global)
+
+  private def poolSettingsWithTimeout():ConnectionPoolSettings = {
+    // We create the pool settings with a timeout: It seems by default, the creation of the settings requires a reverse
+    // dns lookup, which may take more than 5 seconds which triggers the akka streams subscription timeout.
+    // If we let the creation here take the full time, we won't start without any reasonable logging message
+    try {
+      Await.result(poolSettings(), 2.seconds)
```

Comment:
I don't get this. I'm sorry. Why can we not document the fact that apps should increase the Akka stream subscription timeout? It feels we are just introducing complexity here.